src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -1979,6 +1979,44 @@ public void actionPerformed(final ActionEvent e) {
     }
   };
 
+  /**
+   * Prompt user for which producer territory (factory) to use when producing units in sea
+   * zones that have multiple adjacent factories.
+   * 
+   * @param candidates
+   *        - list of producer territories to choose from
+   * @param unitTerritory
+   *        - territory units are being placed
+   * @return territory selected to use for production
```

Comment:
Might want to document that `null` can be returned if the user does not make a selection.

Alternatively, you could return `Optional<Territory>`.